### ISSUE TITLE
Add possibility to remove RequestFilter

### ISSUE THREAD

**Issue by tommywo** [lightbody/browsermob-proxy#326]
Hi, 
I would like to do something like this:
```java
MyRequestFilter myRequestFilter = new MyRequestFilter();
proxy.addRequestFilter(myRequestFilter);
//do something using proxy with myRequestFilter
proxy.removeRequestFilter(myRequestFilter);
// do something using proxy without myRequestFilter
```
Can you add this functionality?

**Comment by jekh** [comment]
I think this is a great idea! It's slightly more complicated than you'd think, because of the way LittleProxy's filters and filter factories work. I've also considered perhaps having the addRequestFilter() method return  a filter ID that can be used to disable or delete the filter later, which would be much friendlier for the REST API.

I think we'll probably get to this when we start modifying the REST API, which will most likely be part of the 2.2 release. In the meantime I'd be happy to entertain any PRs that work within the existing design!

**Comment by d3vw3b** [comment]
+1 

Would love to use this feature as well! :) I'm attempting to test 3 or 4 different web page layouts that are based on user agent. So I'm setting a user agent request filter to change it, going to the site and verifying elements, setting another request filter, going to the site and verifying elements, so on and so forth. However since I can't clear prior request filters at the moment, those are still triggered and I end up with with request filters stacking on top of each other and doing essentially the same thing.

**Comment by chaudharydeepak** [comment]
Hi, first of all thank-you so much for building this awesome tool. I was wondering if this request got implemented? If yes, pardon my negligence. Thank you once again.

**Comment by asaunders-els** [comment]
I don't think this feature has been implemented yet as I have not come across a way to do so via the api. Would anyone know of a work-around? Besides stopping/restarting the proxy.
